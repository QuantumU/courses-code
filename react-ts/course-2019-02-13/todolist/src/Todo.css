ul {
  list-style-type: none;
  padding-left: 0;
}

.todo {
  padding: 0.5rem;
  margin: 0.5rem;
  background-color: lightblue;
}

.todo.completed {
  background-color: lightgrey;
  text-decoration: line-through;
}
