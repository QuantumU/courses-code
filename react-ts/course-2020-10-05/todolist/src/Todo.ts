// Typendeklaration, die in verschiedenen
// Dateien verwendet werden kann

export type Todo = {
  id: number;
  title: string;
  completed: boolean;
};
