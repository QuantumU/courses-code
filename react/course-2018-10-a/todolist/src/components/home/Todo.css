.todo {
  margin: 8px;
  padding: 8px;
  background-color: lightblue;
  cursor: pointer;
}

.todo.completed {
  background-color: lightgrey;
  text-decoration: line-through;
}
