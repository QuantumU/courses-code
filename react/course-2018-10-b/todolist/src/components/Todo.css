.todo {
  margin: 10px;
  padding: 10px;
  background-color: lightblue;
}

.todo.completed {
  background-color: lightgrey;
  text-decoration: line-through;
}
