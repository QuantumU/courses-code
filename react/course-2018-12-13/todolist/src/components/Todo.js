import React, { Component } from 'react';

class Todo extends Component {
  render() {
    return (
      <div onClick={this.handleClick}>
        {this.props.todo.completed ? 'DONE: ' : 'TODO: '}
        {this.props.todo.title}
      </div>
    );
  }

  handleClick = () => {
    // wenn das item angeklickt wird,
    // l√∂se das Event "onCompleted" aus
    this.props.onCompleted(this.props.todo.id);
  };
}

export default Todo;
