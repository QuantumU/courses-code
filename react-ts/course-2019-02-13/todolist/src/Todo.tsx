import React from 'react';
import TodoInterface from './TodoInterface';
import './Todo.css';
import App from './App';

interface TodoProps {
  todo: TodoInterface;
  onToggle: (id: number) => void;
  onDelete: (id: number) => void;
}

function Todo(props: TodoProps) {
  return (
    <li
      className={props.todo.completed ? 'todo completed' : 'todo'}
      onClick={() => {
        props.onToggle(props.todo.id);
      }}
    >
      <input type="checkbox" checked={props.todo.completed} readOnly={true} />
      {props.todo.completed ? 'DONE: ' : 'TODO: '}{props.todo.title}
      <button
        onClick={() => {
          props.onDelete(props.todo.id);
        }}
      >
        X
      </button>
    </li>
  );
}

export default Todo;
