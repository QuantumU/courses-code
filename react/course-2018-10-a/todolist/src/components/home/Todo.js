import React, { PureComponent } from 'react';

import './Todo.css';

class Todo extends PureComponent {
  render() {
    return (
      <div
        key={this.props.todo.id.toString()}
        onClick={() => this.props.onToggle(this.props.todo.id)}
        className={this.props.todo.completed ? 'todo completed' : 'todo'}
      >
        {this.props.todo.completed ? 'DONE:' : 'TODO:'}
        {this.props.todo.title}
      </div>
    );
  }
}

export default Todo;
