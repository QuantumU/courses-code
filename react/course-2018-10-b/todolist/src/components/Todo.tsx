import * as React from 'react';
import ITodo from 'src/ITodo';

// Zusatzfeature von Webpack
import './Todo.css';

interface ITodoProps {
  todo: ITodo;
  toggle: (id: number) => void;
}

class Todo extends React.Component<ITodoProps> {
  public render() {
    return (
      <div
        className={this.props.todo.completed ? 'todo completed' : 'todo'}
        onClick={this.handleClick}
      >
        {this.props.todo.completed ? 'DONE:' : 'TODO:'}
        {this.props.todo.title}
      </div>
    );
  }

  private handleClick = () => {
    this.props.toggle(this.props.todo.id);
  };
}

export default Todo;
