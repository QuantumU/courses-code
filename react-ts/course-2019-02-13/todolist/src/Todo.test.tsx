import React from 'react';
import Todo from './Todo';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Todo
        todo={{ id: 1, completed: false, title: 'test' }}
        onToggle={() => {}}
        onDelete={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
