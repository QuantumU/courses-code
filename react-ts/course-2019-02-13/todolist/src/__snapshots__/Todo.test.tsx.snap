// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  className="todo"
  onClick={[Function]}
>
  <input
    checked={false}
    readOnly={true}
    type="checkbox"
  />
  TODO: 
  test
  <button
    onClick={[Function]}
  >
    X
  </button>
</li>
`;
